### ISSUE TITLE
Review Shadrikov's reproduction status

### ISSUE THREAD

**Issue by vitalwarley** [vitalwarley/research#24]
**Aims and Methods**

- Metrics
  - Baseline
    - AUC (Area Under ROC Curve) = 0.7183
  - Baseline + classification
    - AUC = 0.7762
  - Baseline + classification + normalization
    - AUC = 0.7912
- Model
  - Baseline -- ResNet pretrained on MS1M_v2 (5.8M images, 86k classes) in a classification task.
  - Baseline + classification -- baseline fine tuned on FIW in a classification task.
  - Baseline + classification + normalization -- baseline fine tuned on FIW in a classification task with a L2 normalization of the embeddings.
- How AUC was computed?
  - ResNet pretrained backbone was used to compute embeddings from Families In the Wild (FIW) validation set. Then, cosine distance between the embeddings was used to verify kinship. Distances were then used to compute AUC.

**What did I do?**

- Implement ResNet101 with ArcFace loss;
- Pretrain model on MS1M_v2 dataset;
- Validate results on LFW, CFP_FP, AGEDB_30;

**What did I got?**

- Validation checks on LFW
  - LFW accuracy
    - ArcFace R100 ONNX (my model): 0.8585
    - ArcFace R100 Torch (my model): >0.99.
    - ArcFace R100 MXNET: 0.987
  - I did get the same LFW AUC as Shadrikov on LFW
    - ArcFace R100 ONNX (my model): 0.9985
    - ArcFace R100 MXNET: 0.9992
- Baseline checks on FIW
  - I didn't get the same baseline AUC as Shadrikov on FIW validation set.
      - **ArcFace R100 Torch (my model): 0.6717**
      - **ArcFace R100 ONNX (my model): 0.6717**
      - ArcFace R100 MXNET: 0.7183
  - Both models on both approaches gave an accuracy of 0.63. 

**Current issues**

- Using `arcface_torch` (pre-trained model equivalent to **ArcFace R100 MXNET**) I got AUC = 0.7198 (vs 0.6717 from my pre-trained model) and ACC = 0.66 (same with mxnet).

> We can conclude two things: 1) something is wrong with my model and/or its training; 2) Shadrikov's accuracy seems to be really 0.66, not 0.70, as in the paper.
> This is my fault, though. That accuracy seems to be the weighted average from the accuracies for each relation type. I will confirm that with https://github.com/vitalwarley/research/issues/19.
>
> _Originally posted by @vitalwarley in https://github.com/vitalwarley/research/issues/18#issuecomment-1173724030_

> After #15 and #13, I improved my results. AUC on FIW went to 0.6717 and accuracy went to 0.63. They are less than Shadrikov's model: 0.7183 and 0.67, respectively.
> 
> _Originally posted by @vitalwarley in https://github.com/vitalwarley/research/issues/18#issuecomment-1186194339_
> 

            
> I think two things remain to be tested, but not now because I will focus on [these issues](https://github.com/vitalwarley/research/issues/18#issuecomment-1177986568).
> 
> 1. Use insightface's resnet101 instead of timm's (#11).
> 2. Use PartialFC (#7).
> 
> I couldn't find any other explanation. My pieces of evidence are: 1) insightface's PyTorch backbone gives the [same results](https://github.com/vitalwarley/research/issues/18#issuecomment-1173724030) as the MXNet; 2) #13 -- learning rate curve and loss curve are similar to insightface's.
> 
> _Originally posted by @vitalwarley in https://github.com/vitalwarley/research/issues/18#issuecomment-1186195498_

**Comment by vitalwarley** [comment]
Results from baseline + classification scheme https://github.com/vitalwarley/research/issues/20#issuecomment-1194017526

My results

- lr: 0.000015 (step 49449)
- train/accuracy: 0.878334 (step 49449)
- train/loss: 2.396770 (step 49449)
- val/accuracy: 0.679900 (step 49449)
- val/auc: 0.747001 (step 49449)
- val/threshold: 0.060000 (step 49449)

MXNet

- lr: 0.000000 (step 49)
- train/accuracy: 0.937998 (step 49)
- train/loss: 0.957331 (step 49)
- val/accuracy: 0.708900 (step 49)
- val/roc/auc: 0.775784 (step 49)
- val/threshold: 0.110000 (step 49)

> From the above results, we can conclude:
> - `batch-size` was less than what Shadrikov said it to be (16 vs 48). For his script, only bs=16 resulted in a good train accuracy.
> - `gradient_clip_val` needed to be higher than what Shadrikov said -- I used 5 because that was the value I pretrained the model on MS1M. Also, he said 1.5 on paper but wrote 1.0 on code.
> - My results were close to Shadrikov's. I think my model would improve on train metrics (classification) with more epochs, but maybe not on validation metrics (verification).
> 
> Training with torch or mxnet requires different hparams, so I don't think I can reproduce 100% of his work. Besides that, maybe there are some differences in how torch or mxnet optimizes the model... I don't know.
> 
> _Originally posted by @vitalwarley in https://github.com/vitalwarley/research/issues/20#issuecomment-1194025608_
